internal/kubernetes/pvc.go

```diff
@@ -63,6 +64,34 @@ func ListAllPVCWithStorageclass(client *k8s.Clientset, scName string) (*[]corev1
 	return pl, nil
 }
 
+func ListSinglePVCWithStorageclass(client *k8s.Clientset, scName, pvcName, pvcNamespace string) (*[]corev1.PersistentVolumeClaim, error) {
+	pl := &[]corev1.PersistentVolumeClaim{}
+	listOpt := v1.ListOptions{}
+	ctx := context.TODO()
+
+	pvc, err := client.CoreV1().PersistentVolumeClaims(pvcNamespace).List(ctx, listOpt)
```

Comment:
Instead of listing the PVCs, `Get()` will directly give you the PVC, then no need to iterate over the PVCs to match the storage class or the name.
```suggestion
	pvc, err := client.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(ctx, pvcName)
```

Reply:
this is intentional as we are using PVC list here https://github.com/ceph/persistent-volume-migrator/blob/61f74d4998d67d40857d360ce667682000319daa/cmd/flexToCsi.go#L93

I have done it in this way just to use the existing code...we can change for a single PVC but it will lot of code duplicacy.

Reply:
What code duplication will exist? This change will get rid of the `for` loop below, I don't see duplication.

Reply:
we do multiple operations inside `for` loop and we are using the loop to iterate over multiple PVC. For, single PVC if we don't want to use for loop we'll have to copy all the operations that we have in the loop.

Reply:
ahh, I mixed with other loops in other files. Thanks, I'll change accordingly